src/transformers/generation/utils.py

```diff
@@ -960,7 +960,9 @@ def _get_stopping_criteria(
         if generation_config._eos_token_tensor is not None:
             criteria.append(EosTokenCriteria(eos_token_id=generation_config._eos_token_tensor))
         if (
-            generation_config.assistant_confidence_threshold is not None
+            generation_config.is_assistant is not None
+            and generation_config.is_assistant
```

Comment:
```suggestion
            generation_config.is_assistant
```
`generation_config.is_assistant is not None and generation_config.is_assistant` can be simplified as the latter implies the former